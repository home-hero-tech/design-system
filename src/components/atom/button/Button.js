import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './Button.module.scss';

/* <Button type="primary" container="outlined" shape="squared" size="" /> */

const Button = ({ submit, shape, size, container, type, children }) => {
  const btnType = submit ? 'submit' : 'button';

  const classes = classNames({
    [css['c-btn']]: true,
    [css[`c-btn--${type}`]]: true,
    [css[`c-btn--${shape}`]]: true,
    [css[`c-btn--${container}`]]: true,
    [css[`c-btn--size-${size}`]]: true
  });

  return (
    <button type={btnType} className={classes}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  type: PropTypes.oneOf([
    'default',
    'primary',
    'secondary',
    'danger',
    'success'
  ]),
  submit: PropTypes.bool,
  size: PropTypes.oneOf([1, 2, 3, 4]),
  container: PropTypes.oneOf(['default', 'outlined', 'text']),
  shape: PropTypes.oneOf(['squared', 'rounded'])
};

Button.defaultProps = {
  submit: false,
  type: 'default',
  container: 'default',
  shape: 'rounded',
  size: 2
};

export default Button;
