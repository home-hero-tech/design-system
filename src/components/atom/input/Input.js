import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import css from './Input.module.scss';

const Input = ({
  id,
  name,
  type,
  placeholder,
  value,
  onChange,
  disabled,
  dark
}) => {
  const isCheckOrRadio = type === 'checkbox' || type === 'radio';

  const classes = classNames({
    [css['c-input']]: true,
    [css['c-input--disabled']]: disabled,
    [css[`c-input__${type}`]]: isCheckOrRadio,
    [css[`c-input__${type}--disabled`]]: disabled && isCheckOrRadio,
    [css['c-input--dark']]: dark
  });

  return (
    <input
      className={classes}
      id={id}
      name={name}
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      disabled={disabled}
    />
  );
};

Input.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  disabled: PropTypes.bool,
  dark: PropTypes.bool
};

Input.defaultProps = {
  id: null,
  name: null,
  onChange: f => f,
  placeholder: null,
  type: 'text',
  value: null,
  disabled: null,
  dark: null
};

export default Input;
