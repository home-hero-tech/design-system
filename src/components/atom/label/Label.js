import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import css from './Label.module.scss';

const Label = ({ name, small, children }) => {
  const classes = classNames({
    [css['c-label--small']]: !!small
  });

  return (
    <label className={classes} htmlFor={name}>
      {children}
    </label>
  );
};

Label.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.array]).isRequired,
  name: PropTypes.string,
  small: PropTypes.bool
};

Label.defaultProps = {
  name: null,
  small: false
};

Label.displayName = 'Label';

export default Label;
