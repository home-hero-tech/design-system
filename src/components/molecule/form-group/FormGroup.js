import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Col } from 'react-flexbox-grid';

import css from './FormGroup.module.scss';

const FormGroup = ({ error, icon, success, children, ...otherProps }) => {
  const classes = classNames({
    [css['c-form-group']]: true,
    [css['c-form-group--error']]: error,
    [css['c-form-group--error--icon']]: icon,
    [css['c-form-group--success']]: success
  });

  return (
    <Col {...otherProps} className={classes}>
      {children}
    </Col>
  );
};

FormGroup.propTypes = {
  error: PropTypes.bool,
  icon: PropTypes.bool,
  success: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};
FormGroup.defaultProps = {
  error: false,
  icon: false,
  success: false,
  children: null
};

export default FormGroup;
