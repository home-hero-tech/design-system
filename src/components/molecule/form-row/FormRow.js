import React from 'react';
import PropTypes from 'prop-types';
import { Row } from 'react-flexbox-grid';

import css from './FormRow.module.scss';

const FormRow = ({ children, ...otherProps }) => (
  <Row {...otherProps} className={css['c-form-row']}>
    {children}
  </Row>
);

FormRow.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object
  ]).isRequired
};

export default FormRow;
