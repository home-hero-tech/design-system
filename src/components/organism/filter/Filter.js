import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFilter, faSort } from '@fortawesome/pro-light-svg-icons';
import { Row, Col } from 'react-flexbox-grid';

import css from './Filter.module.scss';

const Filter = ({ tabIndex, onClick, showFilters, totalItems, children }) => {
  const classes = classNames({
    [css['c-filter__content']]: true,
    [css['c-filter__content--open']]: showFilters
  });

  return (
    <div className={css['c-filter']}>
      <div
        className={css['c-filter__header']}
        onClick={onClick}
        role="button"
        tabIndex={tabIndex}
        onKeyPress={() => {}}
      >
        <Row>
          <Col md={2}>
            <FontAwesomeIcon icon={faFilter} />
            <span>Filtro</span>
          </Col>
          <Col md>
            <p>
              {totalItems}
              encontrados
            </p>
          </Col>
          <Col md={2}>
            <FontAwesomeIcon icon={faSort} />
            <span>Ordernar: A-z</span>
          </Col>
        </Row>
      </div>
      <div className={classes}>{children}</div>
    </div>
  );
};

Filter.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.string
  ]).isRequired,
  onClick: PropTypes.func,
  showFilters: PropTypes.bool,
  tabIndex: PropTypes.number,
  totalItems: PropTypes.number
};

Filter.defaultProps = {
  onClick: null,
  showFilters: false,
  tabIndex: 0,
  totalItems: 0
};

export default Filter;
