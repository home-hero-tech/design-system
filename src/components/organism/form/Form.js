import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ onSubmit, className, children }) => (
  <form onSubmit={onSubmit} className={className} noValidate>
    {children}
  </form>
);

Form.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.string
  ]).isRequired,
  onSubmit: PropTypes.func.isRequired,
  className: PropTypes.string
};

Form.defaultProps = {
  className: null
};

export default Form;
