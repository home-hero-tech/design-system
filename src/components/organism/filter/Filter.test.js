import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Grid } from 'react-flexbox-grid';
import Filter from './Filter';
import Form from '../form/Form';
import FormRow from '../../molecule/form-row/FormRow';
import FormGroup from '../../molecule/form-group/FormGroup';
import FormActions from '../../molecule/form-actions/FormActions';
import Label from '../../atom/label/Label';
import Input from '../../atom/input/Input';
import Button from '../../atom/button/Button';

describe('Filter', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <Filter showFilters onClick={() => {}} totalItems={3}>
        <Form onSubmit={() => {}}>
          <Grid fluid>
            <FormRow>
              <FormGroup md={4}>
                <Label name="ipt">Label</Label>
                <Input
                  id="ipt"
                  name="ipt"
                  type="text"
                  placeholder="Placeholder text"
                  value=""
                />
              </FormGroup>
              <FormGroup md={4}>
                <Label name="ipt">Label</Label>
                <Input
                  id="ipt"
                  name="ipt"
                  type="text"
                  placeholder="Placeholder text"
                  value=""
                />
              </FormGroup>
            </FormRow>
            <FormActions alignEnd>
              <Button>Limpar filtros</Button>
              <Button success type="submit">
                Filtrar
              </Button>
            </FormActions>
          </Grid>
        </Form>
      </Filter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
