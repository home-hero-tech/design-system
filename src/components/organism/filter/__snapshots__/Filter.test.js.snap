// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match with the snapshot 1`] = `
<div
  className="c-filter"
>
  <form
    onSubmit={null}
  >
    <Form
      className={null}
      onSubmit={[Function]}
    >
      <Grid
        fluid={true}
      >
        <FormRow>
          <FormGroup
            error={false}
            icon={false}
            md={4}
            success={false}
          >
            <Label
              name="ipt"
            >
              Label
            </Label>
            <Input
              dark={null}
              defaultValue={null}
              disabled={null}
              id="ipt"
              name="ipt"
              onChange={[Function]}
              parentRef={null}
              placeholder="Placeholder text"
              type="text"
              value=""
            />
          </FormGroup>
          <FormGroup
            error={false}
            icon={false}
            md={4}
            success={false}
          >
            <Label
              name="ipt"
            >
              Label
            </Label>
            <Input
              dark={null}
              defaultValue={null}
              disabled={null}
              id="ipt"
              name="ipt"
              onChange={[Function]}
              parentRef={null}
              placeholder="Placeholder text"
              type="text"
              value=""
            />
          </FormGroup>
        </FormRow>
        <FormActions
          alignBetween={false}
          alignEnd={true}
        >
          <Button
            container="default"
            disabled={false}
            loading={false}
            onClick={null}
            shape="rounded"
            size={2}
            submit={false}
            type="default"
          >
            Limpar filtros
          </Button>
          <Button
            container="default"
            disabled={false}
            loading={false}
            onClick={null}
            shape="rounded"
            size={2}
            submit={false}
            success={true}
            type="submit"
          >
            Filtrar
          </Button>
        </FormActions>
      </Grid>
    </Form>
  </form>
</div>
`;
